jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/SyncServerlessAppAction.kt

```diff
@@ -0,0 +1,132 @@
+import com.intellij.openapi.application.runInEdt
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.util.text.SemVer
+import icons.AwsIcons
+import kotlinx.coroutines.runBlocking
+import software.aws.toolkits.jetbrains.core.coroutines.getCoroutineBgContext
+import software.aws.toolkits.jetbrains.core.credentials.AwsConnectionManager
+import software.aws.toolkits.jetbrains.core.credentials.getConnectionSettingsOrThrow
+import software.aws.toolkits.jetbrains.core.executables.ExecutableInstance
+import software.aws.toolkits.jetbrains.core.executables.ExecutableManager
+import software.aws.toolkits.jetbrains.core.executables.getExecutable
+import software.aws.toolkits.jetbrains.services.lambda.SyncServerlessAppWarningDialog
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamExecutable
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamTemplateFileUtils.retrieveSamTemplate
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamTemplateFileUtils.validateTemplateFile
+import software.aws.toolkits.jetbrains.services.lambda.sam.sync.SyncApplicationRunProfile
+import software.aws.toolkits.jetbrains.services.lambda.sam.sync.SyncServerlessApplicationDialog
+import software.aws.toolkits.jetbrains.services.lambda.sam.sync.SyncServerlessApplicationSettings
+import software.aws.toolkits.jetbrains.settings.SamDisplayDevModeWarningSettings
+import software.aws.toolkits.jetbrains.utils.notifyError
+import software.aws.toolkits.jetbrains.utils.notifyNoActiveCredentialsError
+import software.aws.toolkits.jetbrains.utils.notifySamCliNotValidError
+import software.aws.toolkits.resources.message
+
+class SyncServerlessAppAction(private val codeOnly: Boolean = false) : AnAction(
+    message("serverless.application.sync"),
+    null,
+    AwsIcons.Resources.SERVERLESS_APP
+) {
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.getRequiredData(PlatformDataKeys.PROJECT)
+
+        if (!AwsConnectionManager.getInstance(project).isValidConnectionSettings()) {
+            notifyNoActiveCredentialsError(project = project)
+            return
+        }
+
+        ExecutableManager.getInstance().getExecutable<SamExecutable>().thenAccept { samExecutable ->
+            if (samExecutable is ExecutableInstance.InvalidExecutable || samExecutable is ExecutableInstance.UnresolvedExecutable) {
+                notifySamCliNotValidError(
+                    project = project,
+                    content = (samExecutable as ExecutableInstance.BadExecutable).validationError
+                )
+                return@thenAccept
+            }
+
+            val execVersion = SemVer.parseFromText(samExecutable.version) ?: error(" SAM CLI version could not detected")
+            val minVersion = SemVer("1.53.0", 1, 53, 0)
+            val minVersionForUseContainer = SemVer("1.57.0", 1, 57, 0)
```

Comment:
is this specifically for `--use-container` with sync?

Reply:
Yes!